plumbing/format/packfile/decoder_test.go

```diff
@@ -1,4 +1,4 @@
-package packfile
+package packfile_test
```

Comment:
It may be nice to begin the test files with constructor of the 4 (or more) kinds of Decoder where are going to need:
- seekable-transactioner
- seekable-non-transactioner
- no-seekable-non-transactioner
- non-seekable-non-transactioner

Just for general clarity and sanity. It will help a lot with the testing, simplicity and coverage.